src/bin/pgcopydb/ld_transform.c

```diff
@@ -1159,7 +1163,18 @@ parseMessage(StreamContext *privateContext, char *message, JSON_Value *json)
 				}
 			}
 
-			(void) streamLogicalTransactionAppendStatement(txn, stmt);
+			if (canCoalesceLogicalTransactionStatement(txn, stmt))
+			{
+				if (!coalesceLogicalTransactionStatement(txn, stmt))
+				{
+					/* errors have already been logged */
+					return false;
+				}
+			}
+			else
+			{
+				(void) streamLogicalTransactionAppendStatement(txn, stmt);
+			}
```

Comment:
It looks to me like this code should be found within the function `streamLogicalTransactionAppendStatement`.